onnxruntime/python/tools/tensorrt/perf/benchmark.py

```diff
@@ -984,82 +944,94 @@ def run_onnxruntime(args, models):
+                    update_fail_model_map(model_to_fail_ep, name, ep, 'runtime error', e)
                     continue
-
-                logger.info("start to inference {} with {} ...".format(name, ep))
-                logger.info(sess.get_providers())
-
-                if sess:
-                    logger.info("Model inputs nodes:")
-                    for input_meta in sess.get_inputs():
-                        logger.info(input_meta)
-                    logger.info("Model outputs nodes:")
-                    for output_meta in sess.get_outputs():
-                        logger.info(output_meta)
-
-                batch_size = 1
-                result_template = {
-                    "engine": "onnxruntime",
-                    "version": onnxruntime.__version__,
-                    "device": ep,
-                    "fp16": fp16,
-                    "io_binding": False,
-                    "model_name": name,
-                    "inputs": len(sess.get_inputs()),
-                    "batch_size": batch_size,
-                    "sequence_length": 1,
-                    "datetime": str(datetime.now()),}
-                    
-                if trt in ep and args.trtexec:
-                    
-                    # get standalone TensorRT perf
+                
+                # memory tracking variables 
+                p = None # keep track of process to kill upon error
+                mem_usage = None
+         
+                # get standalone TensorRT perf
+                if standalone_trt in ep and args.trtexec: 
+                    trtexec = True 
                     try: 
-                        ep = standalone_trt_fp16 if fp16 else standalone_trt
-                        
                         if args.track_memory: 
                             p = start_memory_tracking()            
                             result = run_trt_standalone(args.trtexec, model_path, sess.get_inputs(), all_inputs_shape, fp16)
-                            mem_usage = end_memory_tracking(p, True)
-                            if result and mem_usage: 
-                                result["memory"] = mem_usage
-
+                            mem_usage = end_memory_tracking(p, trtexec, True)
                         else: 
                             result = run_trt_standalone(args.trtexec, model_path, sess.get_inputs(), all_inputs_shape, fp16)
                     except Exception as e: 
                         logger.error(e)
+                        if args.track_memory:
+                            end_memory_tracking(p, trtexec, False)
                         update_fail_model_map(model_to_fail_ep, name, ep, 'runtime error', e)
                         continue
 
-                else: 
-                    result = inference_ort(args, name, sess, ep, inputs, result_template, args.test_times, batch_size)
-                
+                # inference with onnxruntime ep
+                else:         
```

Comment:
same code as before (aside slight refactor of terminating memory tracking) just moved to else clause because standalone operates as the if